include/sparrow/data_type.hpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+namespace sparrow
+{
+    enum class data_type
```

Comment:
Should we mention that not [all types](https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings) are supported, yet?